test/spec/components/List.spec.js

```diff
@@ -0,0 +1,778 @@
+          label: 'foo'
+        }
+      ];
+
+      const {
+        container,
+        rerender
+      } = createListEntry({ container: parentContainer, items });
+
+      const header = domQuery('.bio-properties-panel-list-entry-header', container);
+
+      const list = domQuery('.bio-properties-panel-list-entry', container);
+
+      waitFor(async () => {
+        await header.click();
+      });
+
+      // assume
+      expect(getListOrdering(list)).to.eql([
+        'item-2',
+        'item-3',
+        'item-1'
+      ]);
+
+      items[2].label = 'aaa';
+
+      // when
+      createListEntry({ items }, rerender);
+
+      // then
+      expect(getListOrdering(list)).to.eql([
+        'item-2',
+        'item-3',
+        'item-1'
+      ]);
+    });
+
+
+    it('complex (open -> add -> change -> remove -> close -> open)', async function() {
+
+      // given
+      let items = [
+        {
+          id: 'item-1',
+          label: 'xyz'
+        },
+        {
+          id: 'item-2',
+          label: 'abc'
+        }
+      ];
+
+      const {
+        container,
+        rerender
+      } = createListEntry({ container: parentContainer, items });
+
+      const header = domQuery('.bio-properties-panel-list-entry-header', container);
+
+      const list = domQuery('.bio-properties-panel-list-entry', container);
+
+      // when
+
+      // (1) open
+      waitFor(async () => {
+        await header.click();
+      });
+
+      expect(getListOrdering(list)).to.eql([
+        'item-2',
+        'item-1'
+      ]);
+
+      // (2) add
+      items = [
+        ...items,
+        {
+          id: 'item-3',
+          label: 'foo'
+        }
+      ];
+
+      createListEntry({ items }, rerender);
+
+      expect(getListOrdering(list)).to.eql([
+        'item-3',
+        'item-2',
+        'item-1'
+      ]);
+
+      // (3) change
+      items[0].label = 'aaa';
+
+      createListEntry({ items }, rerender);
+
+      expect(getListOrdering(list)).to.eql([
+        'item-3',
+        'item-2',
+        'item-1'
+      ]);
+
+      // (4) remove
+      items.splice(1, 1);
+
+      createListEntry({ items }, rerender);
+
+      expect(getListOrdering(list)).to.eql([
+        'item-3',
+        'item-1'
+      ]);
+
+      // (5) close + open
+      waitFor(async () => {
+        await header.click();
+      });
+
+      waitFor(async () => {
+        await header.click();
+      });
+
+      expect(getListOrdering(list)).to.eql([
+        'item-1',
+        'item-3'
+      ]);
+    });
+  });
+
+
+  it('should remove items', function() {
+
+    // given
+    const items = [
+      {
+        id: 'item-1',
+        label: 'Item 1'
+      },
+      {
+        id: 'item-2',
+        label: 'Item 2'
+      },
+      {
+        id: 'item-3',
+        label: 'Item 3'
+      }
+    ];
+
+    const {
+      container,
+      rerender
+    } = createListEntry({ container: parentContainer, items });
+
+    items.splice(0, 1);
```

Comment:
Maybe we can test the usage of a `onRemove` handler here? A `rerender` would probably be needed as well because the items change, but at least we can verify the `onRemove` handler works as expected.

Reply:
OK I will do that.

Reply:
Done.